clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -1222,6 +1222,16 @@ static IsTupleLike isTupleLike(Sema &S, SourceLocation Loc, QualType T,
   if (E.isInvalid())
     return IsTupleLike::Error;
 
+  if (Size < 0 || Size >= UINT_MAX) {
```

Comment:
Why is this `UINT_MAX` but the calling code uses `Ctx.getSizeType()`?

Reply:
The host compiler uintmax is the right thing, because we pass this down on an UnsignedOrNone, which uses unsigned as the representation, so the implementation limit is uintmax-1.